### DISCUSSION TITLE
VIPS DZSAVE

### DISCUSSION THREAD

**Discussion by ferninoa** [libvips/libvips#2770] [Q&A]
Hi, I need help, I am getting a message ("no matches found: --suffix=.jpg[Q=100]") when running vips dzsave command like this:

```
vips dzsave 09216_00-1.tif 09216_00-1 --overlap=0 --tile-width=224 --tile-height=224 --suffix=.jpg[Q=100]
zsh: no matches found: --suffix=.jpg[Q=100]
```

This work fine with vips dzasave version 8.9.2 in an Intel Mac but when I run it in a Mac M1 ARM, vips dzsave version 8.12.2, I get this message: `no matches found: --suffix=.jpg[Q=100]`

If I removed the --suffix it runs and create .jpeg files but not sure what is the default quality?

Thanks,
Fernando

**Comment by jcupitt** [comment]
Hi @ferninoa,

This is being cased by zsh, the new shell on macOS. zsh uses square brackets for patterns, so it's trying to expand your `--suffix=.jpg[Q=100]`.

On zsh, you need to quote things like this. Try:

```
vips dzsave 09216_00-1.tif 09216_00-1 --overlap=0 --tile-width=224 --tile-height=224 "--suffix=.jpg[Q=100]"
```

**Reply by ferninoa** [reply]
Hi John,

Perfect, that works fine. Thank you very much for your prompt answer. VIPS DZSAVE flys on the M1 Mac. 

I have another question, when I run this command "vips dzsave 09216_00-1.tif 09216_00-1 --overlap=0 --tile-width=224 --tile-height=224 "--suffix=.jpg[Q=100]"", I got 16 folders (0-15) created with tiles, which are fine. For my project I used folder 13 only, my images are all the same size so I always get the same results, 400 tiles per image (224x224px) on folder 13. Is there an option to tell VIPS DZSAVE to tile the images create for folder 13 only? Just for the purpose to save time and storage space.

Thanks again for your help.

Have a great day.
Fernando.

**Reply by jcupitt** [reply]
You can use `--depth one` and it'll only save the highest res layer.

If you shrink by x4 before starting the dzsave, you'll get level 13 as the highest res layer.

Something like:

```
vips resize 09216_00-1.tif "09216_00-1.dz[overlap=0,tile-width=224,tile-height=224,suffix=.jpg[Q=100],depth=one]" 0.25
```

**Comment by ferninoa** [comment]
Well, this is amazing, it took a few seconds to create the tiles and it matched with the ones I previously created for the folder/level/layer 13. Thank you very much for the help.

Now, I am curious how you calculate this, what is the relation of x4 (0.25) with level 13. Sometimes I used folder 12 or 14 instead of 13, so I will like to do the same for these cases, just create folders 12 or 14.

**Reply by jcupitt** [reply]
Each level is a x2 reduction, so dropping two levels is (1 / (2 * 2)), or 0.25.

If you want something smarter, you could calculate yourself from the input image size. It sounds like you're aiming for eg. a 20k x 20k pixel image, so you could use `20000 / image-width` as the resize factor.

**Reply by ferninoa** [reply]
John, thank you so much, big help.
